deps/rabbit/src/rabbit_reader.erl

```diff
@@ -1119,7 +1119,17 @@ handle_input({frame_payload, Type, Channel, PayloadSize}, Data, State) ->
 handle_input(handshake, <<"AMQP", A, B, C, D, Rest/binary>>, State) ->
     {Rest, version_negotiation({A, B, C, D}, State)};
 handle_input(handshake, <<Other:8/binary, _/binary>>, #v1{sock = Sock}) ->
-    refuse_connection(Sock, {bad_header, Other});
+    Reason = case Other of
+                 <<16#16, 16#03, _Ver2, _Len1, _Len2, 16#01, _, _>> ->
+                     %% Looks like a TLS client hello.
+                     detected_unexpected_tls_header;
+                 <<"GET ", _URL/binary>> ->
+                     %% Looks like an HTTP request.
+                     detected_unexpected_http_header;
```

Comment:
Nothing important, but I wonder if it might get confusing considering there are HTTP *headers* in HTTP *messages*, and while this refers to the AMQP connection "header" people tend to think of "HTTP header" as the headers in HTTP messages.

Reply:
You're right, the log message should be clearer. https://github.com/rabbitmq/rabbitmq-server/pull/13563 provides a follow up with very clear log messages.